HISTORY.md

```diff
@@ -4,6 +4,10 @@
 * Fixed a bug in handling file rename error in distributed/network file systems when the server succeeds but client returns error. The bug can cause CURRENT file to point to non-existing MANIFEST file, thus DB cannot be opened.
 * Fixed a bug where ingested files were written with incorrect boundary key metadata. In rare cases this could have led to a level's files being wrongly ordered and queries for the boundary keys returning wrong results.
 * Fixed a data race between insertion into memtables and the retrieval of the DB properties `rocksdb.cur-size-active-mem-table`, `rocksdb.cur-size-all-mem-tables`, and `rocksdb.size-all-mem-tables`.
+* Fixed the false positive alert when recovering from the WAL file. Avoid reporting "SST file is ahead of WAL" on a newly created empty column family, if the previous WAL file is corrupted.
+
+### Behavior Changes
+* Due to the fix of false postive alert of "SST file is ahead of WAL", all the CFs with no SST file (CF empty) will bypass the consistency check.
```

Comment:
I think we should mention the potential inconsistency scenario here: an empty CF and the WAL's last a few entries containing its tombstones are corrupted. Therefore, the DB can only recover to the point without these trailing tombstones, while the cf's log number may point to the next WAL.
We fixed a false-positive, but introduced a rare true-negative.

Reply:
Thanks for the review, I will update it accordingly to state the potential risk with more details.

Reply:
> > I think we should mention the potential inconsistency scenario here: an empty CF and the WAL's last a few entries containing its tombstones are corrupted. Therefore, the DB can only recover to the point without these trailing tombstones, while the cf's log number may point to the next WAL.
> > We fixed a false-positive, but introduced a rare true-negative.
> 
> I think the condition is stricter than that. It has to flush to an SST file, and a compaction triggered which combines this file and all other SST files and generate empty data set, or there is another reason to write a manifest entry for this CF after a flush that generates no SST file from an empty CF.

Let me reorganize the trigger condition and add to the HISTORY.md.